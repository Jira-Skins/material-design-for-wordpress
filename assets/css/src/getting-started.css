@import "@material/typography/dist/mdc.typography.css";
@import "@material/layout-grid/dist/mdc.layout-grid.css";
@import "@material/button/dist/mdc.button.css";
@import "./gsm/index.css";
