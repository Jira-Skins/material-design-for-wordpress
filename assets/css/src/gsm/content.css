.material-gsm__content {
	padding-right: 1rem;
}
