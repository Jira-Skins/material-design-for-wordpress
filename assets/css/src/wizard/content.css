.material-wizard__content {
	background-color: #fff;
	box-shadow: 0 2px 2px rgba(0, 0, 0, 0.15);
	padding: 2.5rem;

	& p {
		font-family: var(--mdc-typography-body1-font-family, var(--mdc-typography-font-family, Roboto,));
		font-size: var(--mdc-typography-body1-font-size, 1rem);
		line-height: var(--mdc-typography-body1-line-height);
		font-weight: var(--mdc-typography-body1-font-weight, 400);
		letter-spacing: var(--mdc-typography-body1-letter-spacing, 0.03125em);
		text-decoration: var(--mdc-typography-body1-text-decoration, inherit);
		text-transform: var(--mdc-typography-body1-text-transform, inherit);
	}

	& hr {
		margin: 2.5rem auto;
		width: 80%;
	}

	& .material-wizard__title {
		margin-top: 0;
	}

	& .material-wizard__card img {
		width: 100%;
	}
}
