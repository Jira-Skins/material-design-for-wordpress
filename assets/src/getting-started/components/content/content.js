export * from './wizard';
export * from './theme';
export * from './demo';
export * from './editor';
export * from './blocks';
