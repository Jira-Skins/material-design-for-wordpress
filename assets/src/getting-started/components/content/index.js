const Content = () => {
	return (
		<div className="material-gsm__content mdc-layout-grid__cell mdc-layout-grid__cell--span-9">
			<h2 className="material-gsm__content-title mdc-typography--headline6">
				Onboarding Wizard
			</h2>
			<p className="material-gsm__content-description">
				Curabitur mauris erat congue posuere lacus torquent et convallis,
				parturient rutrum cras senectus augue penatibus malesuada rhoncus,
				ultricies finibus pretium arcu sagittis aliquet nisi.
			</p>
			<div className="material-gsm__content-actions"></div>
		</div>
	);
};

export default Content;
