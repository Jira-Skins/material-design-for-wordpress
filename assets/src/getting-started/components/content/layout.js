import { Fragment } from '@wordpress/element';
import { __ } from '@wordpress/i18n';
import Button from '../../../wizard/components/navigation/button';

export const Layout = props => {
	return (
		<Fragment>
			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Posts Settings', 'material-theme-builder' ) }
			</h2>
			<p className="material-gsm__content-description">
				{ __(
					'Select how posts display on your site using cards or image lists in a wide or normal width layout. Card display options include raised or outlined cards, show or hide post comments, author, excerpt, and date.',
					'material-theme-builder'
				) }
			</p>
			<p className="material-gsm__content-description">
				{ __(
					'Choose how your comment fields display within each post, options include outlines of filled text fields.',
					'material-theme-builder'
				) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Site Navigation (Top app bar)', 'material-theme-builder' ) }
			</h2>
			<p className="material-gsm__content-description">
				{ __(
					'Choose tabs, navigation drawer, or both for your site navigation. Give users the ability to search your site on every page directly from within the top app bar. By default the top app bar will hide when scrolling up, choose “Fixed” to keep the app bar in place above all content.',
					'material-theme-builder'
				) }
			</p>

			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Site Navigation (Footer)', 'material-theme-builder' ) }
			</h2>
			<p className="material-gsm__content-description">
				{ __(
					'Add footer text and give your users a quick way to jump back to the top. Looking to add widgets? Add widgets to the bottom of your site in Widgets settings.',
					'material-theme-builder'
				) }
			</p>

			<div className="material-gsm__content-actions">
				<Button
					style="mdc-button--raised"
					text={ __( 'Customize', 'material-theme-builder' ) }
					trailingIcon="navigate_next"
					link={ mtbGsm.customize }
				/>
			</div>

			<div style={ { height: '20px' } }></div>
		</Fragment>
	);
};
