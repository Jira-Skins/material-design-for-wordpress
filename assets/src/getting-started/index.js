/**
 * WordPress dependencies
 */
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import GettingStarted from './components';

render( <GettingStarted />, document.getElementById( 'material-gsm' ) );
