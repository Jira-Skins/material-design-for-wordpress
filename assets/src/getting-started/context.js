/**
 * Setups context to be used across the app
 *
 */
import { useReducer, createContext } from '@wordpress/element';
import { STATUS } from '../wizard/constants';
import { reducer } from './reducer';
import { TABS } from './constants';

const TabContext = createContext();

const { Provider } = TabContext;

const tabs = Object.keys( TABS );

/**
 * Default state of the world
 *
 */
const initialState = {
	activeTab: 'THEME',
	completed: [ 'WIZARD' ],
	status: STATUS.IDLE,
	actionToInstall: null,
	error: {},
	themeStatus: 'install',
	contentStatus: 'install',
	tabs,
};

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const TabProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default TabContext;
