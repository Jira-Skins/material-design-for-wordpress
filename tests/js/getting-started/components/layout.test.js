/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { Layout } from '../../../../assets/src/getting-started/components/content/layout';
import { TabProvider } from '../../../../assets/src/getting-started/context';

const setup = () => {
	return render(
		<TabProvider>
			<Layout />
		</TabProvider>
	);
};

describe( 'GSM: Layout', () => {
	beforeAll( () => {
		global.mtbGsm = {
			themeStatus: 'ok',
			contentStatus: 'ok',
		};
	} );

	afterEach( cleanup );

	it( 'matches snapshot', () => {
		const wrapper = setup();
		expect( wrapper ).toMatchSnapshot();
	} );
} );
