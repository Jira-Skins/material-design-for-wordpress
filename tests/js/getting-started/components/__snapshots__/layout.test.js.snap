// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GSM: Layout matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        Posts Settings
      </h2>
      <p
        class="material-gsm__content-description"
      >
        Select how posts display on your site using cards or image lists in a wide or normal width layout. Card display options include raised or outlined cards, show or hide post comments, author, excerpt, and date.
      </p>
      <p
        class="material-gsm__content-description"
      >
        Choose how your comment fields display within each post, options include outlines of filled text fields.
      </p>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        Site Navigation (Top app bar)
      </h2>
      <p
        class="material-gsm__content-description"
      >
        Choose tabs, navigation drawer, or both for your site navigation. Give users the ability to search your site on every page directly from within the top app bar. By default the top app bar will hide when scrolling up, choose “Fixed” to keep the app bar in place above all content.
      </p>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        Site Navigation (Footer)
      </h2>
      <p
        class="material-gsm__content-description"
      >
        Add footer text and give your users a quick way to jump back to the top. Looking to add widgets? Add widgets to the bottom of your site in Widgets settings.
      </p>
      <div
        class="material-gsm__content-actions"
      >
        <button
          class="mdc-button mdc-button--raised"
        >
          <span
            class="mdc-button__label"
          >
            Customize
          </span>
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon trailing-icon"
          >
            navigate_next
          </i>
        </button>
      </div>
      <div
        style="height: 20px;"
      />
    </div>
  </body>,
  "container": <div>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      Posts Settings
    </h2>
    <p
      class="material-gsm__content-description"
    >
      Select how posts display on your site using cards or image lists in a wide or normal width layout. Card display options include raised or outlined cards, show or hide post comments, author, excerpt, and date.
    </p>
    <p
      class="material-gsm__content-description"
    >
      Choose how your comment fields display within each post, options include outlines of filled text fields.
    </p>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      Site Navigation (Top app bar)
    </h2>
    <p
      class="material-gsm__content-description"
    >
      Choose tabs, navigation drawer, or both for your site navigation. Give users the ability to search your site on every page directly from within the top app bar. By default the top app bar will hide when scrolling up, choose “Fixed” to keep the app bar in place above all content.
    </p>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      Site Navigation (Footer)
    </h2>
    <p
      class="material-gsm__content-description"
    >
      Add footer text and give your users a quick way to jump back to the top. Looking to add widgets? Add widgets to the bottom of your site in Widgets settings.
    </p>
    <div
      class="material-gsm__content-actions"
    >
      <button
        class="mdc-button mdc-button--raised"
      >
        <span
          class="mdc-button__label"
        >
          Customize
        </span>
        <i
          aria-hidden="true"
          class="material-icons mdc-button__icon trailing-icon"
        >
          navigate_next
        </i>
      </button>
    </div>
    <div
      style="height: 20px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
